external/kafka/src/main/scala/org/apache/spark/streaming/kafka/KafkaUtils.scala

```diff
@@ -144,4 +150,116 @@ object KafkaUtils {
     createStream[K, V, U, T](
       jssc.ssc, kafkaParams.toMap, Map(topics.mapValues(_.intValue()).toSeq: _*), storageLevel)
   }
+
+  /** A batch-oriented interface for consuming from Kafka.
+   * Starting and ending offsets are specified in advance,
+   * so that you can control exactly-once semantics.
+   * @param sc SparkContext object
+   * @param kafkaParams Kafka <a href="http://kafka.apache.org/documentation.html#configuration">
+   * configuration parameters</a>.
+   *   Requires "metadata.broker.list" or "bootstrap.servers" to be set with Kafka broker(s),
+   *   NOT zookeeper servers, specified in host1:port1,host2:port2 form.
+   * @param batch Each OffsetRange in the batch corresponds to a
+   *   range of offsets for a given Kafka topic/partition
+   * @param messageHandler function for translating each message into the desired type
+   */
+  def createRDD[
+    K: ClassTag,
+    V: ClassTag,
+    U <: Decoder[_]: ClassTag,
+    T <: Decoder[_]: ClassTag,
+    R: ClassTag] (
+      sc: SparkContext,
+      kafkaParams: Map[String, String],
+      batch: Array[OffsetRange],
+      messageHandler: MessageAndMetadata[K, V] => R
+  ): RDD[R] = {
+    val parts = batch.zipWithIndex.map { case (o, i) =>
+        new KafkaRDDPartition(i, o.topic, o.partition, o.fromOffset, o.untilOffset, o.host, o.port)
+    }.toArray
+    new KafkaRDD[K, V, U, T, R](sc, kafkaParams, parts, messageHandler)
+  }
+
+  /**
+   * This DOES NOT guarantee that side-effects of an action will see each message exactly once.
```

Comment:
Alright, I understand your point of view :) Then lets consider other alternate names. We could simply name it generic `createNewStream`. Which is okay, because eventually, if all performance and other issues are flattened out, we would like to graduate it from "experimental feature" (Spark 1.3 will mark it experimental) to the default supported implementation of kafka stream. So "new stream" may be a natural indicator about which one to use.